RecodoDesktop/Recodo.Desktop.Logic/RecorderService.cs

```diff
@@ -0,0 +1,140 @@
+            source.Add(Recorder.GetWindows().FirstOrDefault(w => w.Title == _options.RecorderWindowTitle));
+
+            var options = new RecorderOptions
+            {
+                AudioOptions = new AudioOptions
+                {
+                    AudioInputDevice = Recorder.GetSystemAudioDevices(AudioDeviceSource.InputDevices)
+                    .FirstOrDefault(d => d.FriendlyName == _options.SelectedAudioInputDevice)?
+                    .DeviceName,
+
+                    AudioOutputDevice = Recorder.GetSystemAudioDevices(AudioDeviceSource.OutputDevices)
+                    .FirstOrDefault(d => d.FriendlyName == _options.SelectedAudioOutputDevice)?
+                    .DeviceName,
+
+                    IsAudioEnabled = _options.IsAudioEnable,
+                    IsInputDeviceEnabled = true,
+                    IsOutputDeviceEnabled = true,
+                },
+                SourceOptions = source.FirstOrDefault() == null 
+                ? SourceOptions.MainMonitor 
+                : new SourceOptions { RecordingSources = source },
+                OutputOptions = new OutputOptions
+                {
+                    RecorderMode = RecorderMode.Video,
+                    OutputFrameSize = new ScreenSize(_options.Resolution.Item1, _options.Resolution.Item2),
+                    Stretch = StretchMode.Uniform,
+                }
+            };
+
+            recorder = Recorder.CreateRecorder(options);
+            recorder.OnRecordingFailed += Rec_OnRecordingFailed;
+            recorder.OnRecordingComplete += Rec_OnRecordingComplete;
+            recorder.OnStatusChanged += Rec_OnStatusChanged;
+            recorder.Record(Path.ChangeExtension(Path.GetTempFileName(), ".mp4"));
+
+            StartRec?.Invoke();
+        }
+
+        public void PauseRecording()
+        {
+            recorder?.Pause();
+        }
+
+        public void ResumeRecording()
+        {
+            if (recorder?.Status == RecorderStatus.Paused)
+            {
+                recorder?.Resume();
+            }
+        }
+
+        public void StopRecording()
+        {
+            recorder?.Stop();
+        }
+
+        public List<string> GetInputAudioDevices()
+        {
+            return Recorder.GetSystemAudioDevices(AudioDeviceSource.InputDevices)
+                ?.Select(d=>d.FriendlyName)
+                ?.ToList();
+        }
+
+        public List<string> GetOutputAudioDevices()
+        {
+            return Recorder.GetSystemAudioDevices(AudioDeviceSource.OutputDevices)
+                ?.Select(d => d.FriendlyName)
+                ?.ToList();
+        }
+
+        public List<RecordingSourceBase> GetDisplays()
+        {
+            var sources = new List<RecordingSourceBase>();
+            sources.AddRange(Recorder.GetDisplays());
+
+            return sources;
+        }
+
+        public List<string> GetWindows()
+        { 
+            return Recorder.GetWindows().Select(w => w.Title).ToList();
```

Comment:
Can't they have the same titles? If I open two windows of chrome

Reply:
So there will be the same two titles but it will be different objects